filter/series_by_tag.go

```diff
@@ -223,14 +216,39 @@ func createMatchingHandlerForOneTag(
 	}, nil
 }
 
+func handleRegexMatch(
+	isMatchOperator bool,
+	spec TagSpec,
+	compatibility *Compatibility,
+) (func(string) bool, error) {
+	// We don't need to create a regular for the asterisk, because in such a tag
+	// it is the fact of the tag's presence that matters, not its value.
+	if spec.Value == "*" || spec.Value == ".*" {
+		return func(value string) bool {
+			return isMatchOperator
+		}, nil
+	}
+
+	matchRegex, err := newMatchRegex(spec.Value, compatibility)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new match regex: %w", err)
+	}
+
+	return func(value string) bool {
+		match := matchRegex.MatchString(value)
+
+		if isMatchOperator {
```

Comment:
Мб стоит упростить булево выражение?

```suggestion
		return isMatchOperator == match
```

Reply:
Я вот думал над этим, но как будто сложность восприятия кода возрастает. Давай сделаю, как предложил, просто комментом отпишу